app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt

```diff
@@ -154,9 +159,114 @@ class LocalSource(
                 }
             }
 
+        // get Metadata from ComicInfo.xml Files
+        val metadataRegex = "(no)?xml".toRegex()
+        val metadataFile = getMangaDirsFiles(manga.url, baseDirsFile)
+            .firstOrNull { it.extension.matches(metadataRegex) }
+
+        try {
+            val encoding = "UTF-8"
+            if (metadataFile?.extension == "xml") {
+                val comicInfo = xml.decodeFromReader<ComicInfo>(AndroidXmlReader(metadataFile.inputStream(), encoding))
```

Comment:
Need to close the reader/stream, if this doesnt work use `.use` on the `inputStream()` instead
```suggestion
                val comicInfo = AndroidXmlReader(metadataFile.inputStream(), encoding).use { xml.decodeFromReader<ComicInfo>(it) }
```